lib/foreman_scap_client/client.rb

```diff
@@ -36,7 +36,14 @@ def scan_command
                                else
                                  ''
                                end
-      "oscap xccdf eval #{fetch_remote_resources} #{profile} #{tailoring_subcommand} --results-arf #{results_path} #{config[@policy_id][:content_path]}"
+      # OpenSCAP 1.3.6 and newer requires the `--local-files` option to use local copies of remote SDS components
+      version = `rpm -q openscap`.split('-')[1]
```

Comment:
Also, even though I understand that this is expected to have `openscap` installed to be able to run any scan commands this gem provides, what if `rpm -q openscap` returns `package openscap is not installed`? This will eventually raise something like `undefined method '>=' for nil:NilClass` due to `version >= '1.3.6'`. Not so helpful error in case there is no such package. I'm not familiar with this gem much, but I see there is some error handling: https://github.com/theforeman/foreman_scap_client/blob/a4245cb2317e60e328321100af85bf7a620e9ece/lib/foreman_scap_client/base_client.rb#L81-L105

Reply:
I think it's a good point that you also want to suppress stderr output from the command.

Reply:
Looking further at this, I'd recommend the command `rpm -q --qf '%{version}' openscap` since that only prints the version and there's no need to split it in Ruby. Then as @ofedoren suggested I'd use the `Open3.capture3` method.
```suggestion
      version, _stderr, status = Open3.capture3('rpm', '-q', '--qf', '%{version}', 'openscap')
      return false unless status.success?

```

You can also combine it into a single comparison `status.success? &&       Gem::Version.new(version) >= Gem::Version.new('1.3.6')` but I don't know if the line will be too long then.